### PR TITLE
[1.1] Accelerate sparse embedding grad op in CPU device

### PR DESCRIPTION
**Pull request by velconia**
Save 84.75% time for grad op in CPU model.

Save around 5% time for forward op in CPU model.

### CODE REVIEW

File: paddle/fluid/operators/lookup_table_op.cc [hunk 1]
```diff
@@ -81,6 +81,10 @@ class LookupTableOpMaker : public framework::OpProtoAndCheckerMaker {
                      "Otherwise the given value indicates padding the output "
                      "with zeros whenever lookup encounters it in Ids.")
         .SetDefault(kNoPadding);
+    AddAttr<bool>("grad_inplace",
+                  "(boolean, default false) "
+                  "If the grad op reuse the input's variable.")
```
Comment (panyx0718) [inline paddle/fluid/operators/lookup_table_op.cc:88]: Tell people don't use it?

**Review by kuke — APPROVED** [review]
LGTM

File: paddle/fluid/operators/lookup_table_op.h [hunk 2]
```diff
@@ -111,27 +113,37 @@ class LookupTableGradKernel : public framework::OpKernel<T> {
...
-          d_table_value->dims(),
-          framework::flatten_to_2d(d_output_dims, d_output_dims.size() - 1));
-      memcpy(d_table_data, d_output_data, sizeof(T) * d_output->numel());
+      // FIXME(minqiyang):
+      // memory optimization will NOT reuse Tensor with SelectedRows
+      // so we could just share the tensor here directly.
+      // However, the InferVarType method will infer the output SelectedRows
+      // to Tensor sometimes, which is a bug, so we will add an attribute
+      // here to indicate the inplace and remove this attribute after
+      // the InferVarType's bug was fixed
```
Comment (jacquesqiao) [inline paddle/fluid/operators/lookup_table_op.h:129]: CUDA kernel can also support inplace.

Comment (jacquesqiao) [reply paddle/fluid/operators/lookup_table_op.h:129]: need unit test

**Pull request merged**